zilliqa/src/db.rs

```diff
@@ -337,6 +325,23 @@ impl Db {
             return Err(anyhow!("Invalid checkpoint file: parent's blockhash does not correspond to checkpoint block"));
         }
 
+        if !trie_storage.db.is_empty() || self.get_highest_block_number()?.is_some() {
+            // This may not be strictly necessary, as in theory old values will, at worst, be orphaned
+            // values not part of any state trie of any known block. With some effort, this could
+            // even be supported.
+            // However, without such explicit support, having old blocks MAY in fact cause
+            // unexpected and unwanted behaviour. Thus we currently forbid loading a checkpoint in
+            // a node that already contains previous state, until (and unless) there's ever a
+            // usecase for going through the effort to support it and ensure it works as expected.
+            if let Some(db_block) = self.get_block_by_hash(&block.hash())? {
+                if db_block.parent_hash() != block.parent_hash() {
```

Comment:
Should we just check `db_block != block` instead?

Reply:
Yeah, that probably would have been more sensible